### DISCUSSION TITLE
Background Uploads of multiple large Files

### DISCUSSION THREAD

**Discussion by urm-dev** [firebase/firebase-ios-sdk#10751] [Q&A]
Hi all,

I implemented an app that uploads multiple big files containing measurement data (around 100MB per file). For my use case it is important that the files are uploaded sequentially. Currently, the app needs to run in foreground until all files are uploaded.

I am looking for a way to continue uploading files with the Firebase iOS SDK when the app moves to the background / is suspended (e.g. the user locks the screen or switches to a different app).

Is there any way to achieve this with the current version of the Firebase iOS SDK? I have found this feature request that was opened in 2017, but activity stopped in the end fo 2021:
https://github.com/firebase/firebase-ios-sdk/issues/147

I couldn't find anything either in the documentation about background uploads:
https://firebase.google.com/docs/storage/ios/upload-files

Thanks in advance for any inputs!

**Comment by xaphod** [comment]
I just commented on #147 again - what most people don't realize is that background sessions are being created, which breaks any framework that wants to manage its own retries.... but you can't actually _use_ them for anything useful, because there's no FIRStorage API thru GTMSessionFetcher to actually hook up your rehydration code etc.
Very frustrating.
